src/transport_none.h

```diff
@@ -55,10 +55,9 @@ typedef int shm_internal_datatype_t;
 typedef int shm_internal_op_t;
 typedef int shmem_transport_ct_t;
 
-struct shmem_transport_ctx_t{};
+struct shmem_transport_ctx_t{ int dummy };
 
 typedef struct shmem_transport_ctx_t shmem_transport_ctx_t;
-shmem_transport_ctx_t shmem_transport_ctx_default;
```

Comment:
Depending on how we define SHMEM_CTX_DEFAULT in the public header, we will likely need to add a ```transport_none.c``` file to define a symbol for it.  In the prototype, I think we declared ```extern void * SHMEM_CTX_DEFAULT``` in ```shmem.h``` and then defined ```shmem_transport_ctx_t SHMEM_CTX_DEFAULT``` in the .c file for each transport.